### PR TITLE
docker: add Makefiles for easier docker deploy

### PR DESCRIPTION
**Pull request by RandyMcMillan**

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+all: init
+	@echo 'all'
+	docker-compose up --force-recreate --always-recreate-deps
+	@echo ''
+#######################
+.PHONY: torproxy
+torproxy:
+	@echo 'torproxy'
+#REF: https://hub.docker.com/r/dperson/torproxy
+	bash -c 'docker run -it -p 8118:8118 -p 9050:9050 -p 9051:9051 -d dperson/torproxy'
```
Comment (wiz) [inline GNUmakefile:59]: Why is this needed? It seems to be not used for anything, and it adds a trusted third party dependency.

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    command: "./wait-for-it.sh db:3306 --timeout=720 --strict -- ./start.sh"
+    volumes:
+      - ./data:/backend/cache
+    environment:
+      RPC_HOST: "127.0.0.1"
+      RPC_PORT: "8332"
+      RPC_USER: "mempool"
+      RPC_PASS: "mempool"
+      ELECTRUM_HOST: "127.0.0.1"
+      ELECTRUM_PORT: "50002"
```
Comment (wiz) [inline docker/docker-compose.yml:29]: It seems this assumes you already have romanz/electrs running on the host machine. Maybe we should add romanz/electrs to this docker-compose template, perhaps as a fragment option?

File: GNUmakefile (hunk 1 above)
Comment (RandyMcMillan) [reply GNUmakefile:59]: I will take it out :)

File: docker/docker-compose.yml (hunk 2 above)
Comment (RandyMcMillan) [reply docker/docker-compose.yml:29]: thanks for the feed back - i will add this.

Comment (RandyMcMillan) [reply docker/docker-compose.yml:29]: We also want to avoid conflicts with other bundled docker implementations (like umbrel) so it will be an optional command in make - i will add some notes in documentation as well.

File: .gitignore [hunk 3]
```diff
@@ -1 +1,6 @@
 sitemap
+data
+docker-compose.yml
+mysql
```
Comment (wiz) [inline .gitignore:4]: Is this needed? Looks like `mysql` will go in `data/`

File: docker/electrs/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,34 @@
+ARG VERSION=v0.8.6
```
Comment (wiz) [inline docker/electrs/Dockerfile:1]: Looks like the most recent release is v0.8.9, any reason not to use that instead?

File: .gitignore [hunk 5]
```diff
@@ -1 +1,6 @@
...
+electrs
```
Comment (wiz) [inline .gitignore:5]: Also looks like `electrs` will go in `data/` unless I'm missing something?

**Review by wiz — APPROVED** [review]
utACK
